### PR TITLE
fix: Allow empty value in select type widgets options property

### LINKED ISSUE #10007: [Bug]: should allow empty string as value in array type properties in multiple widgets
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Due to our recent property [validation related fixes](https://github.com/appsmithorg/appsmith/pull/9476), Multiple widgets are not allowing empty string as values. This was working before and now it's breaking some of the user's applications.

- [x] Checkbox group widget 
- [x] MultiSelecttree widget
- [x] Multiselect widget
- [x] dropdown widget
- [x] radio group widget
- [x] single select tree
- [ ] switch group widget

### Steps To Reproduce

1. Add a single select widget to canvas (or anyone from the list above)
2. Add following array as options `[{"label": "option1", "value": ""}, {"label": "option2", "value": "2"}]`
3. Error is shown saying that invalid value.

Expected:
Should accept empty string `""` as the value in **one** of the option.
### Environment

Production

### Version

Cloud

### PR DESCRIPTION
**Pull request by sbalaji1192**
Fixes #10007 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce.
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/select-widget-empty-value 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.09 **(0.01)** | 36.53 **(0.01)** | 34.48 **(0)** | 55.61 **(0)**
 :green_circle: | app/client/src/utils/WorkerUtil.ts | 89.76 **(0.78)** | 72.55 **(1.96)** | 100 **(0)** | 93.33 **(0.95)**</details>

### CODE REVIEW

**Comment by YogeshJayaseelan** [comment]
@sbalaji1192 Tested this PR and please find the status below,

Currently one option value can be empty and this is now applicable to **Select Widget**, **Swtich Group Widget**, **Checkbox Group** and **MultiSelect Widget** 

**Issue 1** - Radio group accepts multiple empty value and no validation message is displayed 

https://www.loom.com/share/a6e7f7d7c6284852b27074bbe066a58b

**Question 1** - For **Multi SelectTree** and **Single TreeSelect** - Can we have one empty value in parent level and one empty value in child level? Currently if there is one empty value in parent and one empty value in child - validation error occurs and Not sure if this is a valid case. 

https://loom.com/share/dc7e01e1938a4b9393c190c763a7c402

**Comment by sbalaji1192** [comment]
```Question 1 - For Multi SelectTree and Single TreeSelect - Can we have one empty value in parent level and one empty value in child level? Currently if there is one empty value in parent and one empty value in child - validation error occurs and Not sure if this is a valid case.```
@somangshu what should be the expected behaviour here?

**Comment by somangshu** [comment]
@sbalaji1192 The question is do we allow empty / null values in the options. If thats the case I would say all the value and have the unique validation in place, This however might have created some issue in the past which I might not be aware of. 

@dilippitchika could you check this once, What is the right behavior?

**Comment by sbalaji1192** [comment]
@somangshu @dilippitchika The question is do we need to allow empty values at different levels in multi select tree options?

**Comment by dilippitchika** [comment]
Discussed with @sbalaji1192 on the issue. We should only allow one null/empty value across the tree of options in multi-tree-select and tree-select. The users are using this to manage "None" option behavior on their side and this can be easily managed by having only one option for "None" throughout the tree.

**Comment by sbalaji1192** [comment]
@YogeshJayaseelan Fixed issue 1 - Now radio group widget enforces unique value. please verify.

**Comment by YogeshJayaseelan** [comment]
@sbalaji1192 Verified the fix and it is working as expected. We can proceed further

File: app/client/src/widgets/RadioGroupWidget/widget/index.tsx [hunk 1]
```diff
@@ -26,6 +26,7 @@ class RadioGroupWidget extends BaseWidget<RadioGroupWidgetProps, WidgetState> {
               type: ValidationTypes.ARRAY,
               unique: ["value"],
               params: {
+                unique: ["value"],
```
Comment (jsartisan) [inline app/client/src/widgets/RadioGroupWidget/widget/index.tsx:29]: QQ: what does `unique: ["value"]` mean which is there one line above?

**Review by jsartisan — APPROVED** [review]

File: app/client/src/widgets/RadioGroupWidget/widget/index.tsx (hunk 1 above)
Comment (sbalaji1192) [reply app/client/src/widgets/RadioGroupWidget/widget/index.tsx:29]: I think it's a misplacement. removing it.

**Review by jsartisan — APPROVED** [review]

**Pull request merged**